### ISSUE TITLE
Atlas

### ISSUE THREAD

**Issue by PyTimGit** [nilearn/nilearn#1496]
Hi everyone, 
I stumbled across nilearn a few days ago. Great tool and a teriffic way to harness the power of sklearn. Thanks!
Now, I'm trying to get all voxel values from a structural MRI (VBM) image for every ROI in a given atlas. What I would like to get out is one np array for every ROI (or a list of arrays or whatever works). After trying for an hour with NiftiMasker, NiftiLabelsMasker, and NiftiMapsMasker, it seems as though nilearn 
1) can only take 4d images (LabelsMasker and MapsMasker)
or 2) cannot handle the atlas format used in the Nilearn (... datasets.fetch_atlas...).
For example, NiftiMasker takes binary masks without a problem, however, I would have to convert every single ROI from Nilearn's own AAL atlas to be able to use NiftiMasker to extract all ROIs in an atlas, it seems. On the other hand, LabelsMasker and MapsMasker appear to be able to handle Nilearn atlas formats, but do not support extracting voxel data from 3d (i.e. non-resting-state, structural) images. 
Could you please give me a hint on how to solve this? I'm sure I'm overlooking something very simple that would enable using the Nilearn atlases for data extraction in 3d images.
Thanks in advance. Cheers,
Tim

**Comment by KamalakerDadi** [comment]
Please refer to user guide thoroughly to know usage with the Nilearn.

NiftiLabelsMasker and NiftiMapsMasker are meant to extract signals from 4D nifti images given the atlas.
Typical case if fMRI images.

What you want is NiftiMasker. Nice overview is given here.
Userguide for masker objects: http://nilearn.github.io/manipulating_images/masker_objects.html

Have a look at this link. I made a small example of what you want.
https://gist.github.com/KamalakerDadi/c36e7435c685a851e48a25b77f021e73

**Comment by PyTimGit** [comment]
Thanks a lot for the quick reply! That's exactly what I meant: I realize that solving this with a loop using NiftiMasker is possible. I just thought there would be a build-in solution. The example you gave is simple enough though. Thanks again.
